src/main/java/seedu/address/model/person/Person.java

```diff
@@ -22,7 +23,7 @@ public class Person {
     private final Email email;
 
     // Data fields
-    private final Address address;
+    private final Optional<Address> address;
     private final Set<Tag> tags = new HashSet<>();
 
     /**
```

Comment:
The comment on the constructor here says that every field must be present and not null. At the start of the constructor, there is also a `requireAllNonNull` that includes the `address` param, so it might be a bit weird in this case

Two ideas:
1. Remove `address` from `requireAllNonNull`. Then I would think that the `address` parameter to this constructor should be `Optional<Address>`.
2. Create a new constructor that does not have `address` as a parameter